src/components/portrait/portrait-initials.component.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Browser from '../../utils/helpers/browser';
+
+class PortraitInitials extends React.Component {
+  static propTypes = {
+    /** A custom CSS class for the component. */
+    className: PropTypes.string,
+
+    /** The user's initials to render. */
+    initials: PropTypes.string.isRequired,
+
+    /** The dimensions (size) of the canvas, in pixels. */
+    dimensions: PropTypes.number.isRequired,
+
+    /** Use a dark background. */
+    darkBackground: PropTypes.bool,
+
+    /** The `alt` HTML string. */
+    alt: PropTypes.string
+  }
+
+  static defaultProps = {
+    darkBackground: false
+  }
+
+  /** Cache of the initials graphic. */
+  cachedImageDataUrl = null;
+
+  /** Invoked before a mounted component receives new props. */
+  componentWillReceiveProps(nextProps) {
+    const shouldClearCache = (
+      this.props.initials !== nextProps.initials
+      || this.props.dimensions !== nextProps.dimensions
+      || this.props.darkBackground !== nextProps.darkBackground
+    );
+
+    if (shouldClearCache) {
+      this.cachedImageDataUrl = null;
+    }
+  }
+
+  /** Generates a graphic with initials. */
+  generateDataUrl() {
+    if (this.cachedImageDataUrl) {
+      return this.cachedImageDataUrl;
+    }
+
+    let canvas = Browser.getDocument().createElement('canvas');
+    let context = canvas.getContext('2d');
+
+    // Set canvas with & height
+    canvas.width = this.props.dimensions;
+    canvas.height = this.props.dimensions;
+
+    // Select a font family to support different language characters
+    // like Arial
+    context.font = `${Math.round(canvas.width / 2.4)}px Lato, Arial`;
+    context.textAlign = 'center';
+
+    // Setup background and front color
+    context = this.applyBackground(context);
+    context = this.applyText(context);
+
+    // Set image representation in default format (png)
+    const dataURI = canvas.toDataURL();
+
+    // Dispose canvas element
+    canvas = null;
+
+    this.cachedImageDataUrl = dataURI;
+
+    return this.cachedImageDataUrl;
+  }
+
+  /** Applies the background colour to the canvas. */
+  applyBackground(canvasContext) {
+    const color = this.props.darkBackground ? '#8A8E95' : '#D8D9DC';
```

Comment:
Is the legacy or DLS colors? If it's going to be used in DLS do we need to have hardcoded values? Maybe it would be good to use colors from themes? It should more flexible for changes in the future

Reply:
Good point. In ee8dd75 the DLS colours are now obtained from the theme :thumbsup: